src/crawlee/playwright_crawler/_playwright_crawler.py

```diff
@@ -101,17 +101,37 @@ async def _make_http_request(
         if self._browser_pool is None:
             raise ValueError('Browser pool is not initialized.')
 
-        # Create a new browser page
         crawlee_page = await self._browser_pool.new_page(proxy_info=context.proxy_info)
 
         async with crawlee_page.page:
-            # Navigate to the URL and get response.
-            response = await crawlee_page.page.goto(context.request.url)
+            goto_options = {}
+
+            playwright_context = PlaywrightCrawlingContext(
+                request=context.request,
+                session=context.session,
+                add_requests=context.add_requests,
+                send_request=context.send_request,
+                push_data=context.push_data,
+                proxy_info=context.proxy_info,
+                get_key_value_store=context.get_key_value_store,
+                log=context.log,
+                page=crawlee_page.page,
+                infinite_scroll=lambda: infinite_scroll(crawlee_page.page),
+                response=None,
+                enqueue_links=None,  # We'll set this later
```

Comment:
This is a no-go. First, it won't pass the type checker. Second, nullable attributes lead to errors, mental overhead and hard to read code.

Instead, we should add one more parent class to `PlaywrightCrawlingContext` that only adds the `page` attribute. Then, we should split the `_make_http_request` into two stages - one that opens the page and executes prenavigation hooks, and second that does the navigation. The context pipeline (see line 78) should be responsible for chaining these.

Does that make sense?